x-pack/plugins/observability/public/pages/threshold/types.ts

```diff
@@ -0,0 +1,177 @@
+import { LensPublicStart } from '@kbn/lens-plugin/public';
+import { ObservabilitySharedPluginStart } from '@kbn/observability-shared-plugin/public';
+import { SharePluginStart } from '@kbn/share-plugin/public';
+import { SpacesPluginStart } from '@kbn/spaces-plugin/public';
+import {
+  RuleTypeParams,
+  TriggersAndActionsUIPublicPluginStart,
+} from '@kbn/triggers-actions-ui-plugin/public';
+import { UiActionsStart } from '@kbn/ui-actions-plugin/public';
+import { UnifiedSearchPublicPluginStart } from '@kbn/unified-search-plugin/public';
+import { UsageCollectionStart } from '@kbn/usage-collection-plugin/public';
+import { TimeUnitChar } from '../../../common/utils/formatters';
+import { MetricsExplorerSeries } from '../../../common/threshold_rule/metrics_explorer';
+import {
+  Comparator,
+  CustomMetricExpressionParams,
+  FilterQuery,
+  MetricExpressionParams,
+  MetricsSourceStatus,
+  NonCountMetricExpressionParams,
+  SnapshotCustomMetricInput,
+} from '../../../common/threshold_rule/types';
+import { ObservabilityPublicStart } from '../../plugin';
+import { MetricsExplorerOptions } from './hooks/use_metrics_explorer_options';
+
+export interface AlertContextMeta {
+  currentOptions?: Partial<MetricsExplorerOptions>;
+  series?: MetricsExplorerSeries;
+}
+
+export type MetricExpression = Omit<
+  MetricExpressionParams,
+  'metric' | 'timeSize' | 'timeUnit' | 'metrics' | 'equation' | 'customMetrics'
+> & {
+  metric?: NonCountMetricExpressionParams['metric'];
+  customMetrics?: CustomMetricExpressionParams['customMetrics'];
+  label?: CustomMetricExpressionParams['label'];
+  equation?: CustomMetricExpressionParams['equation'];
+  timeSize?: MetricExpressionParams['timeSize'];
+  timeUnit?: MetricExpressionParams['timeUnit'];
+};
+
+export enum AGGREGATION_TYPES {
+  COUNT = 'count',
+  AVERAGE = 'avg',
+  SUM = 'sum',
+  MIN = 'min',
+  MAX = 'max',
+  RATE = 'rate',
+  CARDINALITY = 'cardinality',
+  P95 = 'p95',
+  P99 = 'p99',
+  CUSTOM = 'custom',
+}
+
+export interface MetricThresholdAlertParams {
+  criteria?: MetricExpression[];
+  groupBy?: string | string[];
+  filterQuery?: string;
+  sourceId?: string;
+}
+
+export interface ExpressionChartRow {
+  timestamp: number;
+  value: number;
+}
+
+export type ExpressionChartSeries = ExpressionChartRow[][];
+
+export interface TimeRange {
+  from?: string;
+  to?: string;
+}
+
+export interface AlertParams {
+  criteria: MetricExpression[];
+  groupBy?: string | string[];
+  filterQuery?: FilterQuery;
+  sourceId: string;
+  filterQueryText?: string;
+  alertOnNoData?: boolean;
+  alertOnGroupDisappear?: boolean;
+  shouldDropPartialBuckets?: boolean;
+}
+
+export interface InfraClientStartDeps {
+  cases: CasesUiStart;
+  charts: ChartsPluginStart;
+  data: DataPublicPluginStart;
+  dataViews: DataViewsPublicPluginStart;
+  discover: DiscoverStart;
+  embeddable?: EmbeddableStart;
+  lens: LensPublicStart;
+  // TODO:: check if needed
```

Comment:
@fkanout Would you please also include the ticket in the TODO comments? :)

Reply:
sure!

Reply:
adding it here too to appear in the issue too https://github.com/elastic/kibana/issues/159340